File: solutions/c/two-fer/1/two_fer.c
#include "two_fer.h"
#include <stdio.h>
#include <string.h>
void two_fer(char *buffer, const char *name){
    if (name && strlen(name)>0) sprintf(buffer, "One for %s, one for me.", name);
    else strcpy(buffer,"One for you, one for me.");
    return;
}

File: solutions/fortran/two-fer/1/two_fer.f90
module two_fer
  implicit none

contains

  function twoFer(name) result(phrase)
    character(*), intent(in), optional :: name
    character(:), allocatable :: phrase

    if (trim(name) == "") then
            phrase = "One for you, one for me."
    else
            phrase = "One for " // name // ", one for me."        
    end if        
  end function twoFer

end module two_fer
